dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java

```diff
@@ -454,27 +454,25 @@ public Entry createAndSetDiscourseTopic(Long id) throws CustomWebApplicationExce
 
         // Create title and link to entry
 
-        String entryLink = "https://dockstore.org/";
-        String title = "";
-        if (hostName.contains("staging")) {
-            entryLink = "https://staging.dockstore.org/";
-            title = "Staging ";
-        }
+        final String entryPath;
+        final String sitePath;
         if (entry instanceof BioWorkflow) {
-            title += ((BioWorkflow)(entry)).getWorkflowPath();
-            entryLink += "workflows/";
+            entryPath = ((BioWorkflow)entry).getWorkflowPath();
+            sitePath = "workflows/" + entryPath;
         } else if (entry instanceof Service) {
-            title += ((Service)(entry)).getWorkflowPath();
-            entryLink += "services/";
+            entryPath = ((Service)entry).getWorkflowPath();
+            sitePath = "services/" + entryPath;
         } else if (entry instanceof AppTool) {
-            title += ((AppTool)(entry)).getWorkflowPath();
-            entryLink += "tools/";
+            entryPath = ((AppTool)entry).getWorkflowPath();
+            sitePath = "tools/" + entryPath;
         } else {
-            title += ((Tool)(entry)).getToolPath();
-            entryLink += "tools/";
+            entryPath = ((Tool)entry).getToolPath();
+            sitePath = "tools/" + entryPath;
         }
 
-        entryLink += title;
+        final boolean isStaging = hostName.contains("staging");
```

Comment:
wonder if we should throw in qa and dev on the off-chance that we hook up those for some dev/testing at some point
it looks like we're just looking at the portion before `.dockstore`

Reply:
We could append "Staging "-or-equivalent for all platforms that aren't prod (because why not).